app/controllers/staff/transactions_controller.rb

```diff
@@ -59,6 +59,16 @@ def update
     end
   end
 
+  def show
+    @transaction = policy_scope(Transaction).find(id)
+    authorize @transaction
+
+    @activity = Activity.find_by(hierarchy_id: @transaction.fund)
```

Comment:
The reference to `fund` here is a concern as it doesn't support other hierarchies without more logic.

Have we missed that transactions need a polymorphic association to the fund as well, just like activity? That would enable us to use `transaction.hierarchy` in our query instead. If this sounds like the right thing to do could you perhaps put something in Trello so we don't forget, then we can get this merged!

Reply:
Yep looks like we need a polymorphic association here: https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/app/models/transaction.rb#L2